### DISCUSSION TITLE
how to get access to minio from extern

### DISCUSSION THREAD

**Discussion by oswaldB** [Budibase/budibase#2474] [Help] [answered]
**EDIT : This is documented at https://docs.budibase.com/self-hosting/accessing-minio**

Hello,

i need to have access to the minio server to upload files on there. As I cannot use automation to do so, as automation doesn't allow me to upload a file on row creation, I am trying to upload my file with N8N. I can collect the list of the bucket but I need to know a bit more about the buckets,folder architecture to have access to it.

Thanks you

**Comment by oswaldB** [comment]
this is the error I get:
Error: getaddrinfo ENOTFOUND prod-budi-app-assets.SERVERIP

**Comment by shogunpurple** [comment]
Hi @oswaldB  - if you want to connect directly to your MinIO cluster, you should do so directly. The MinIO URL (when running locally with the defaults and using docker compose) will be `http://localhost:4004`. 

You will also need to include your `MINIO_ACCESS_KEY` and `MINIO_SECRET_KEY` in your requests, both of which can be found in your `.env` file or `hosting.properties` file on the server that you installed budibase on.

**Comment by oswaldB** [comment]
Thanks.

What is the structure in terms of buckets? Files?


Thank you

**Comment by shogunpurple** [comment, marked as answer]
@oswaldB 

Each app has it's own key under the bucket - it's appId.

Attachments are in:
`<app-id>/attachments/`

In order for this to work you would also need to update couchDB with the reference to the key of your attachment - this is so budibase knows which attachment records point to which files. You can find the code for this here:

https://github.com/Budibase/budibase/blob/master/packages/server/src/api/controllers/static/index.js

**Comment by oswaldB** [comment]
Thanks a lot.

I will try tomorrow.

Have a nice evening
